### PR TITLE
MSODP-1074 Update for Dart 2 and add a Slack adapter

### PR DESCRIPTION
**Pull request by georgelesica-wf**
This moves bender.dart into the Dart 2 world and brings it up to speed with the current state of Bendium.

### CODE REVIEW

File: lib/src/repository/actions.dart [hunk 1]
```diff
@@ -15,17 +15,17 @@ import 'package:bender/src/repository/update_results.dart';
...
+      getDeployPrAction(),
+      getMergeMasterAction(),
+      getMonitorStatusAction(),
+      getPubGetAction(),
+      getRerunSkynetAction(),
+      getRerunSmithyAction(),
+      getTestConsumersAction(),
+      getUpdateGoldsAction(),
+      getUpdateResultsAction(),
+    ];
```
Comment (charliekump-wf) [inline lib/src/repository/actions.dart:39]: Current Bendium actions:

final Iterable<Action> actions = <Action>[
  createJiraTicket,
  testConsumers,
  deployPR,
  detectConflict, ///Missing
  monitorStatus,
  mergeMaster,
  testBranch, ///Missing
  updateGolds,
  abideUpdate, ///Missing
  dartFormat,
  pubGet,
  runBootstrap,
  rerunBuild,
  bumpVersion,
  cutRelease,  ///Missing?
  updateDartDeps ///Missing
];

File: lib/src/repository/actions.dart [hunk 2]
```diff
@@ -15,17 +15,17 @@ import 'package:bender/src/repository/update_results.dart';
...
+      getDartFormatAction(),
+      getDeployPrAction(),
+      getMergeMasterAction(),
+      getMonitorStatusAction(),
+      getPubGetAction(),
+      getRerunSkynetAction(),
+      getRerunSmithyAction(),
+      getTestConsumersAction(),
+      getUpdateGoldsAction(),
+      getUpdateResultsAction(),
```
Comment (charliekump-wf) [inline lib/src/repository/actions.dart]: Remove this action.

File: lib/src/repository/actions.dart [hunk 3]
```diff
@@ -15,17 +15,17 @@ import 'package:bender/src/repository/update_results.dart';
...
-];
+      getBootstrapAction(),
+      getBumpVersionAction(),
+      getCreateJiraTicketAction(),
+      getDartFormatAction(),
+      getDeployPrAction(),
+      getMergeMasterAction(),
+      getMonitorStatusAction(),
+      getPubGetAction(),
+      getRerunSkynetAction(),
```
Comment (charliekump-wf) [inline lib/src/repository/actions.dart]: Remove this action

File: lib/src/repository/actions.dart (hunk 1 above)
Comment (georgelesica-wf) [reply lib/src/repository/actions.dart:39]: * [x] createJiraTicket,
* [x] testConsumers,
* [x] deployPR,
* [x] detectConflict, ///Missing
* [x] monitorStatus,
* [x] mergeMaster,
* [x] testBranch, ///Missing
* [x] updateGolds,
* [x] abideUpdate, ///Missing
* [x] dartFormat,
* [x] pubGet,
* [x] runBootstrap,
* [x] rerunBuild,
* [x] bumpVersion,
* [x] cutRelease,  ///Missing?
* [x] updateDartDeps ///Missing

File: lib/src/repository/actions.dart (hunk 2 above)
Comment (georgelesica-wf) [reply lib/src/repository/actions.dart]: check

File: lib/src/repository/actions.dart (hunk 3 above)
Comment (georgelesica-wf) [reply lib/src/repository/actions.dart]: check

**Pull request merged**